### DISCUSSION TITLE
I want give to my team specific settings in openvscode server

### DISCUSSION THREAD

**Discussion by jiungbaek** [gitpod-io/openvscode-server#349] [Q&A]
Hi there

I want release to my team specific settings in openvscode server

However, there is no settings.json in openvscode server.

How to make static settings in openvscode server?(using settings.json or something)

**Comment by jeanp413** [comment]
Hi @jiungbaek, are these settings per project? if so you could just create a `settings.json` inside `.vscode` folder in the repo and commit it or are you looking into something like settings sync in vscode desktop where you log in and if you open vscode in a different machine it will sync settings, extensions, etc?

**Reply by jiungbaek** [reply]
i doesn want sync vscode desktop.

where is settings.json? i can't find settings.json

i want shere settings.json with openvscode-server tar.

**Reply by jeanp413** [reply]
There's no settings.json in vscode server user settings are stored in the browser storage

**Reply by jiungbaek** [reply]
code-server  and desktop vscode has settings.json.
why doesn't openvscode-server have settings.json?
If i want share the same setting to many user, i need settings.json

**Reply by jeanp413** [reply]
code-server have a custom patch modifying upstream code for enabling this, the purpose of openvscode-server is to do the minimal changes on upstream code

**Reply by jiungbaek** [reply]
thanks
how do i find my openvscode server settings in browser storage?

**Reply by jeanp413** [reply]
It's stored as bytes in the using Indexed Database API so I don't think it will be of much help, you can open chrome dev tools go to Application > IndexDB > vscode-userdata-store > /User/settings.json

**Reply by jiungbaek** [reply]
Thanks
If  I changed browser (chrome to edge), all setting will reset to default right?

**Reply by jeanp413** [reply]
Yes, that's when settings sync comes into play, if you think that would be useful we recently implemented it for the [gitpod extension](https://github.com/gitpod-io/openvscode-server/pull/337) in vscode desktop and we plan to release for openvscode-server too

**Comment by jiungbaek** [comment]
Is that any way to set default folder except add "?folder=/path" in url?

**Reply by jeanp413** [reply]
There's `--default-folder` and `--default-workspace` server arguments but there's a bug and they are not respected right now, I'll fix it for 1.67.1 this week

**Comment by jiungbaek** [comment]
hi. i have another question.
I can use `~/.openvscode-server/data/Machine/settings.json` file to share default setting to all users.
however, `"window.menuBarVisibility": "classic"` option can not work in Machine/settings.json
How can i share this option?

**Reply by jeanp413** [reply]
@jiungbaek If I understand correctly what you want to do is customize the initial default settings of the workbench, in that case what you should do is add a `configurationDefaults` attribute in the [workbench.ts file](https://github.com/gitpod-io/openvscode-server/blob/c0b47aae888e6f7f70fb11b15f668c32f252148a/src/vs/code/browser/workbench/workbench.ts#L510-L518) with your default settings, here's an example [how we do it in gitpod](https://github.com/gitpod-io/openvscode-server/blob/3dd5bb05411a5eb2a9ded6aca5d38d938fefaf7f/src/vs/gitpod/browser/workbench/workbench.ts#L959-L963) but you'll need to build from sources again with `yarn gulp vscode-reh-web-linux-x64-min`

**Reply by jiungbaek** [reply]
Thanks it works well

**Comment by jiungbaek** [comment]
Hi
If the browser cache is cleared, will all user settings be reset?

**Reply by jeanp413** [reply]
Yes, the settings updated by the user manually will be lost and the default settings in `configurationDefaults` will be used, for this case it's were settings sync is useful

**Reply by jiungbaek** [reply]
you mean `setting sync` will merge to openvscode server and is it will useful?

**Reply by jeanp413** [reply]
Settings sync logic is already part of core VS Code, what's missing is to configure the setting sync server endpoint and a way to authenticate to the setting sync server, this is done by [gitpod extension](https://github.com/gitpod-io/gitpod-vscode-desktop#settings-sync) but right now it's only supported in vscode desktop, I need to make it a web extension so it's also supported by openvscode-server.

**Comment by sullemanhossam** [comment]
hi there im looking for settings sync was this implemented id that so it automatically gets the users settings and applies them
